lesson5/argirovatina_hometask.rb

```diff
@@ -0,0 +1,28 @@
+class LazyHash < Hash
+
+  def lazy?(key)
+    self[key].respond_to? :call
+  end
+
+  def []=(key, value)
+    unless self.respond_to? key
+      define_singleton_method(key) do
+        result = value.respond_to?(:call) ? value.call : value
```

Comment:
I'd rather reassign this _key_ only if value was a block.
I mean in you case it happend anyway, even if the result will be the same.
What do you think about this:
```ruby
value.respond_to?(:call) ? self[key] = value.call : value
```
As you see, if `value` does not respond to `#call`, we just return current _value_, that's it.